### PR TITLE
Responsive navbar

### PR DESCRIPTION
**Pull request by dvitamin**

### CODE REVIEW

File: templates/partials/topnav.html [hunk 1]
```diff
@@ -18,5 +21,141 @@
...
+            <div class="ipopupmenu ipopupmenu--vertical">
+                <ul class="ipopupmenu__list ipopupmenu__list--more">
+                    <!-- prettier-ignore -->
+                    {%- for nav in topnav.children -%}
+                    {% set navPath = nav.path | replace("html", "") %}
+                    {% set navPath = navPath | replace(".", "") %}
+                    {% set activePage = navPath in doc.fileInfo.fullPath %}
+                    <li
+                        class="ipopupmenu__list__item {% if activePage %}ipopupmenu__list__item--highlight{% endif %}"
+                        data-item="nav-item-{{loop.index}}"
```
Comment (ext) [inline templates/partials/topnav.html:42]: I don't see `dataset.item` being accessed?

File: templates/partials/topnav.html [hunk 2]
```diff
@@ -18,5 +21,141 @@
...
+
+<script>
+    function getElementsRefs() {
+        const nav = document.querySelector("#topnav");
+        const moreItem = nav.querySelector(".imenu__list__item--more");
+        const moreButton = moreItem.querySelector("button");
+        const popoverContainer = moreItem.querySelector(".ipopupmenu");
+        const popover = popoverContainer.querySelector(".ipopupmenu__list");
+        const menuItems = Array.from(nav.querySelectorAll(".imenu__list__item[data-item]"));
+        const popoverItems = Array.from(popover.querySelectorAll(".ipopupmenu__list__item[data-item]"));
```
Comment (ext) [inline templates/partials/topnav.html:61]:
```suggestion
        const popoverItems = Array.from(popover.querySelectorAll("li"));
```

Is it not enough to select the `<li>` elements here? Makes for an easier-to-read selector.

File: templates/partials/topnav.html [hunk 3]
```diff
@@ -18,5 +21,141 @@
...
+    }
+
+    function setPopoverVisibility(visible, refs) {
+        refs.popover.style.visibility = visible ? "" : "hidden";
+        refs.moreButton.ariaExpanded = visible.toString();
+    }
+
+    function togglePopover(event, refs) {
+        setPopoverVisibility(refs.popover.style.visibility === "hidden", refs);
+        event.stopPropagation();
```
Comment (ext) [inline templates/partials/topnav.html:75]: Dont know if intentional but if `setPopoverVisibility` fails `event.stopPropagation()` would not be called. Maybe move it so it runs first? Same issue in other places.

File: docs/src/fkui.css [hunk 4]
```diff
@@ -1938,3 +1938,47 @@ code.hljs {
 .imenu--horizontal .imenu__list__anchor-span {
     padding-right: 0.75rem;
 }
+
+.ipopupmenu {
+    background-color: var(--f-background-popupmenu, #ffffff);
+}
+
+.ipopupmenu__list {
```
Comment (ext) [inline docs/src/fkui.css:1946]: Maybe we could use more descriptive class names? Or are we expecting to use more `IPopup` instances?

`.topnav__list`?

**Comment by ext** [comment]
There is also a slight flickering while the page is loading, could it be prevented somehow? Worst-case with javascript.

**Comment by ext** [comment]
With javascript disabled the popover menu is always present and in the way, preventing me from using the topnav properly and covers part of the primary content.

File: templates/partials/topnav.html [hunk 5]
```diff
@@ -18,5 +21,141 @@
...
+                        <a href="{{ nav.path | relative(doc) }}"> {{ nav.title }} </a>
+                    </li>
+                    {%- endfor -%}
+                </ul>
+            </div>
+        </li>
     </ul>
 </nav>
+
+<script>
```
[the remark this answers is not in the document]
Comment (dvitamin) [reply templates/partials/topnav.html:53]: - [ ] move script to runtime

File: templates/partials/topnav.html [hunk 6]
```diff
@@ -18,5 +21,141 @@
...
+            return sum > containerWidth;
+        });
+    }
+
+    function setMenuItemVisibility(itemPair, visible) {
+        itemPair.menu.style.visibility = visible ? "" : "hidden";
+        itemPair.popover.style.display = visible ? "none" : "block";
+    }
+
+    function calculateVisibility(refs) {
```
[the remark this answers is not in the document]
Comment (dvitamin) [reply templates/partials/topnav.html:106]: Issue was no consideration for space between items in overflow calculations. Fixed in 63757d45aead9546ab705927a9fae781ac4d2fd6.

File: templates/partials/topnav.html [hunk 7]
```diff
@@ -18,5 +21,141 @@
...
+
+        const classOperation = refs.highlightIndex >= overflowIndex ? "add" : "remove";
+        refs.moreItem.classList[classOperation]("imenu__list__item--highlight");
+        refs.moreItem.style.left = `${menuItems[overflowIndex].offsetLeft}px`;
+        refs.moreItem.style.visibility = "";
+        const popupTop = refs.moreItem.offsetHeight + 16;
+        refs.popover.style.top = `${popupTop}px`;
+    }
+
+    function onContentReady(callback) {
```
[the remark this answers is not in the document]
Comment (dvitamin) [reply templates/partials/topnav.html:139]: - [ ] use `onContentReady` after move to runtime

File: templates/partials/topnav.html (hunk 1 above)
Comment (dvitamin) [reply templates/partials/topnav.html:42]: - [ ] remove `data-item`, no longer used

File: templates/partials/topnav.html (hunk 2 above)
Comment (dvitamin) [reply templates/partials/topnav.html:61]: I don't want to include the more item.

- [ ] simplify query

File: templates/partials/topnav.html (hunk 3 above)
Comment (dvitamin) [reply templates/partials/topnav.html:75]: - [ ] stop propagation earlier at all places

File: docs/src/fkui.css (hunk 4 above)
Comment (dvitamin) [reply docs/src/fkui.css:1946]: In `fkui.css` I don't know, shouldn't it be a copy of FKUI scss?

File: templates/partials/topnav.html (hunk 1 above)
Comment (dvitamin) [reply templates/partials/topnav.html]: Fixed in ca30434821df95859e9b535e0e440e9e4ad0bbf6.

File: templates/partials/topnav.html (hunk 2 above)
Comment (dvitamin) [reply templates/partials/topnav.html]: Fixed in ca30434821df95859e9b535e0e440e9e4ad0bbf6.

File: templates/partials/topnav.html (hunk 3 above)
Comment (dvitamin) [reply templates/partials/topnav.html:68]: Fixed in 0f3e1e2d4757ab8ea9683d076140a672bc403eaf.

File: templates/partials/topnav.html (hunk 5 above)
[the remark this answers is not in the document]
Comment (dvitamin) [reply templates/partials/topnav.html]: Fixed in 2b7b78e7b74ce545d7c13be72211f1cde2615a9f.

File: templates/partials/topnav.html (hunk 7 above)
[the remark this answers is not in the document]
Comment (dvitamin) [reply templates/partials/topnav.html]: Fixed in 2b7b78e7b74ce545d7c13be72211f1cde2615a9f.

**Comment by dvitamin** [comment]
- [ ] squash before merge

**Comment by dvitamin** [comment]
> With javascript disabled the popover menu is always present and in the way, preventing me from using the topnav properly and covers part of the primary content.

- [x] Hide by default

**Review by MCFK — APPROVED** [review]

**Pull request merged**